### PR TITLE
Streamlines minifiy process

### PR DESCRIPTION
**Pull request by lil5**
**Streamlines dev process**

Just run `npm install && npm run build`.

This generates `./examples/apache-directory-list/header.html`, `./examples/apache-directory-list-local/header.html`, `./examples/bookmarklet/source-min.js` from `./src/source-pp.js` using [Preprocessor](https://www.npmjs.com/package/preprocessor)

**Fixes some issues too**
* #25 .1
* #20

### CODE REVIEW

**Comment by lil5** [comment]
@dom111 any thoughts?

**Comment by dom111** [comment]
Apologies, work has been busy this past week! This looks really good and is fed more robust than my rubbing a script my side! I'll try and take a look through this over the weekend if I don't get to it before. Thank you again for your work on this!

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+end_of_line = lf
+insert_final_newline = true
+
+[{package.json,.travis.yml}]
+indent_style = space
+indent_size = 2
+
+[*.js]
+indent_style = space
+indent_size = 4
```
Comment (dom111) [inline .editorconfig]: Probably worth keeping this consistent with the other files? I have no objections with them being the same really...

Comment (lil5) [reply .editorconfig]: Could you point out which other extensions as using indent rule globally is a bad idea.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "scripts": {
+    "build:pp": "preprocess ./src/source-pp.js > ./tmp/source-local.js && preprocess ./src/source-pp.js -CDN=true > ./tmp/source-cdn.js",
+    "build:uglify": "uglifyjs ./tmp/source-local.js -o ./tmp/source-local-min.js -c -m -e && uglifyjs ./tmp/source-cdn.js -o ./tmp/source-cdn-min.js -c -m -e",
+    "build:bookmarklet": "echo -n 'javascript:' > ./examples/bookmarklet/source-min.js && cat ./tmp/source-cdn-min.js >> ./examples/bookmarklet/source-min.js",
+    "build:apache-cdn": "OUT=./examples/apache-directory-list/header.html; echo -n '<script type=\"text/javascript\">' > $OUT && cat ./tmp/source-cdn-min.js >> $OUT && echo -n '</script><!--' >> $OUT",
+    "build:apache-local": "OUT=./examples/apache-directory-list-local/header.html; echo -n '<script type=\"text/javascript\">' > $OUT && cat ./tmp/source-local-min.js >> $OUT && echo -n '</script><!--' >> $OUT",
+    "build:cleancss": "cleancss -o ./assets/css/style-min.css ./assets/css/style.css",
+    "build": "npm run build:cleancss && npm run build:pp && npm run build:uglify && npm run build:bookmarklet && npm run build:apache-cdn && npm run build:apache-local"
+  },
```
[the remark this answers is not in the document]
Comment (dom111) [reply package.json]: Ahhh, no apologies, I should have specified the filename. I meant `src/webdav.js` > `src/webdav-min.js`!

[the remark this answers is not in the document]
Comment (lil5) [reply package.json]: Ooo, Yes of course

[the remark this answers is not in the document]
Comment (lil5) [reply package.json]: https://github.com/dom111/webdav-js/pull/29/commits/42759d668d19b68e01e8ed4d0377622a63769481 done

File: .editorconfig (hunk 1 above)
Comment (lil5) [reply .editorconfig]: https://github.com/dom111/webdav-js/pull/29/commits/4d1a7b51699e125885522056f5ef32b8d8b629ad done

**Comment by dom111** [comment]
All looks good. Good catch on the apths in the preprocessor too! I'll sort out this merge tomorrow. Thanks again!

**Comment by lil5** [comment]
> All looks good. Good catch on the apths in the preprocessor too! I'll sort out this merge tomorrow. Thanks again!

That's thanks to easy testing tools :wink:

**Pull request merged**